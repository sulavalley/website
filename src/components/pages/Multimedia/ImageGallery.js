import React from 'react';
import { Modal, Form, Input, Button, Image, Segment } from 'semantic-ui-react';
import base from '../../../base';

class ImageGallery extends React.Component {
  state = {
    images: [],
    imageModalIsOpen: false,
    newImage: {
      url: '',
      title: ''
    }
  };

  componentDidMount() {
    base.fetch('images', {
      context: this,
      asArray: true,
      then(data) {
        this.setState({ images: data });
      }
    });
  }

  handleChange = event => {
    const { newImage } = this.state;
    this.setState({
      newImage: { ...newImage, [event.target.name]: event.target.value }
    });
  };

  handleSubmit = () => {
    const { newImage } = this.state;
    base
      .push('images', {
        data: { ...newImage }
      })
      .then(() => this.setState({ imageModalIsOpen: false }))
      .catch(() => console.error('Something went terribly wrong'));
  };

  render() {
    const { imageModalIsOpen, images } = this.state;
    const { currentUser } = this.props;
    return (
      <div>
        {currentUser && (
          <Button onClick={() => this.setState({ imageModalIsOpen: true })}>
            Nueva publicación
          </Button>
        )}
        <Segment basic>
          <Image.Group size="medium">
            {images.map((image, i) => (
              <Image rounded key={i} src={image.url} />
            ))}
          </Image.Group>
        </Segment>
        <ImageModal
          open={imageModalIsOpen}
          onClose={() => this.setState({ imageModalIsOpen: false })}
          onSubmit={this.handleSubmit}
          onChange={this.handleChange}
        />
      </div>
    );
  }
}

const ImageModal = ({ open, onClose, onSubmit, onChange }) => (
  <Modal closeIcon open={open} onClose={onClose}>
    <Modal.Header>Nueva publicación</Modal.Header>
    <Modal.Content>
      <Form onSubmit={onSubmit}>
        <Form.Field>
          <label>Título</label>
          <Input
            onChange={onChange}
            name="title"
            placeholder="Ingrese un título"
          />
        </Form.Field>
        <Form.Field>
          <label>URL</label>
          <Input
            onChange={onChange}
            name="url"
            placeholder="Ingrese un enlace a la imagen"
          />
        </Form.Field>
        <Button type="submit">Publicar</Button>
      </Form>
    </Modal.Content>
  </Modal>
);

export default ImageGallery;
