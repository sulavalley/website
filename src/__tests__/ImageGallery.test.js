import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ImageGallery from '../components/ImageGallery';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ImageGallery />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<ImageGallery />);
});

test('renders correctly', () => {
  const tree = renderer.create(<ImageGallery />).toJSON();
  expect(tree).toMatchSnapshot();
});
