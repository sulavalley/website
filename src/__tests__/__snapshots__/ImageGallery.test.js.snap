// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="image-gallery"
>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <br />
    <div
      className="field"
    >
      <label>
        Nombre de la imagen
      </label>
      <input
        onChange={[Function]}
        placeholder="Nombre de la imagen"
        type="text"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <label>
        Imagen
      </label>
      <input
        onChange={[Function]}
        placeholder="Enlace de la imagen"
        type="url"
        value=""
      />
    </div>
    <button
      className="ui button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
      type="submit"
    >
      Crear
    </button>
  </form>
  <div
    className="ReactGridGallery"
    id="ReactGridGallery"
  >
    <iframe
      style={
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
          "height": 0,
          "margin": 0,
          "overflow": "hidden",
          "padding": 0,
          "position": "fixed",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
